src/structures/buffers.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use crate::tss2_esys::{TPM2_MAX_DIGEST_BUFFER, TPM2_MAX_NV_BUFFER_SIZE};
+use crate::{Error, Result, WrapperErrorKind};
+use log::error;
+use std::convert::TryFrom;
+use std::ops::Deref;
+
+#[allow(unused_macros)]
+macro_rules! buffer_type {
+    ($native_type:ident,$MAX:expr,$tss_type:ident) => {
+        use crate::tss2_esys::$tss_type;
+
+        #[derive(Debug, Clone, PartialEq, Eq, Default)]
+        pub struct $native_type(Vec<u8>);
+        impl $native_type {
+            const MAX_SIZE: usize = $MAX;
```

Comment:
If there is no point having the max sized stored like that in the macro unless we make it public. And I think it should be public in order for user to avoid creating errors.